src/core/ext/client_channel/client_channel_factory.c

```diff
@@ -85,3 +90,120 @@ grpc_arg grpc_client_channel_factory_create_channel_arg(
   arg.value.pointer.vtable = &factory_arg_vtable;
   return arg;
 }
+
+struct grpc_channel_targets_info {
```

Comment:
Agreed. Given that this PR is not high priority but #8842 is and will likely be merged very soon, I'll  start making the changes to use mdstr_hash_table and then s/mdstr/slice. This allows reviews to happen in the meantime and once #8842 is in, this one can shortly follow.